File: frontend-client/src/services/productService.js
import api from './api';

export const getAllProducts = async (page = 0, size = 10) => {
    const response = await api.get('/api/v1/products', {
        params: { page, size }
    });
    return response.data;
};

export const getProductById = async (id) => {
    const response = await api.get(`/api/v1/products/${id}`);
    return response.data;
};

// Admin Routes
export const createProduct = async (productData) => {
    const response = await api.post('/api/v1/products', productData);
    return response.data;
};

export const updateProduct = async (id, productData) => {
    const response = await api.put(`/api/v1/products/${id}`, productData);
    return response.data;
};

export const deleteProduct = async (id) => {
    await api.delete(`/api/v1/products/${id}`);
};

File: product-service/src/main/java/com/smartinventory/product/service/ProductService.java
package com.smartinventory.product.service;

import com.smartinventory.product.dto.request.ProductRequest;
import com.smartinventory.product.dto.response.ProductResponse;
import org.springframework.data.domain.Page;

public interface ProductService {
    ProductResponse createProduct(ProductRequest request);
    ProductResponse updateProduct(Long id, ProductRequest request);
    ProductResponse getProductById(Long id);
    Page<ProductResponse> getAllProducts(int page, int size);
    Page<ProductResponse> getProductsByCategory(Long categoryId, int page, int size);
    void deleteProduct(Long id);
}
